internal/pkg/daemon/bpfrecorder/bpfrecorder_apparmor.go

```diff
@@ -54,17 +54,24 @@ var appArmorHooks = []string{
 
 var pathWithPid *regexp.Regexp = regexp.MustCompile(`^/proc/\d+/`)
 
+// mntnsID is a unique identifier for a group of processes usually running in a container
+// Note: on a host running concurrent containers, there will be multiple process running with
+// the same PID but they are assigned to different mntns since  they run in different containers.
+// Therefore, in order to have unique apparmor profiles, each profile should be recorded using
+// mntns + pid as a key identifier.
```

Comment:
I'm not sure I understand the mntns + pid part. 

My understanding is that right now (with this PR) we record by mntns and disregard the pid, which makes sense to me: Child processes within a container effectively inherit the main process' security profile (and contribute to it during recording). Do we have more advanced plans here? :)

(Nit: mntns + pid isn't strictly unique, we would need pidns + pid :))

Reply:
We introduced mount namespace originally because we had concurrent recording for two different containers running into a cluster. For instance, two different ProfileRecording CRs are creating in simultaneously which triggers two parallel recordings. 

Since the instrumentation is shared, the pidid might be the same for these two containers, but the mount ns is unique per container as far as I am aware of. The combination of two, allowed us to distinguish between the events coming from these two recording when they are sent up by the kernel.

Reply:
> We introduced mount namespace originally because we had concurrent recording for two different containers running into a cluster. For instance, two different ProfileRecording CRs are creating in simultaneously which triggers two parallel recordings.

All makes sense. 👍 

> The combination of two, allowed us to distinguish between the events coming from these two recording when they are sent up by the kernel.

Where are we combining mntns and pid? I thought we only use mntns to distinguish containers. Put differently, it's 

```
recordedCapabilities     map[mntnsID][]int
```
and not 
```
recordedCapabilities     map[mntnsIDAndPid][]int
```

Why does the docstring read "each profile should be recorded using mntns + pid as a key identifier." and not just "each profile should be recorded using mntns as a key identifier."?

(The implementation makes sense to me, I'm only curious why the pid is explicitly mentioned in the docstring :))

Reply:
comment updated